precisely/mapping_matchers.py

```diff
@@ -3,16 +3,21 @@
 from .results import matched, unmatched, indented_list
 
 
-def is_mapping(matchers):
+def is_mapping(matchers, allow_extra=False):
```

Comment:
If we're creating a separate function to allow extra keys, I don't think we should allow the argument in `is_mapping()`.

Reply:
I went with the kwarg for 2 reasons:
- maximize code reuse
- it makes `is_mapping_with` into syntactic sugar around `is_mapping`

Both of those seemed to me to be good reasons for this structure, but if you prefer that I copypaste and modify is_mapping I'll gladly do so.

Reply:
Modified.